react/src/component/login/login.tsx

```diff
@@ -0,0 +1,217 @@
+                    )
+                }</>
+            }>
+                {
+                    loginStatus.displayName && loginStatus.displayName.length > 0 && (
+                        <Dropdown.Header>{loginStatus.displayName}</Dropdown.Header>
+                    )
+                }
+                {
+                    loginStatus.paymentPlan && (
+                        <Dropdown.Header>{appWrapper.getPaymentPlanName(loginStatus.paymentPlan, schema)}</Dropdown.Header>
+                    )
+                }
+                {
+                    loginStatus.userName && (
+                        <Dropdown.Header>{loginStatus.userName}</Dropdown.Header>
+                    )
+                }
+                <Dropdown.Item href={`/${appName}/logout`}>Sign Out</Dropdown.Item>
+            </NavToggleDropdown>
+        )
+    }
+
+    return (
+        <Nav.Link href={`/${appName}/login`}>Sign In</Nav.Link>
+    )
+
+
+}
+
+export const CatchLoggedOut = (props) => {
+    let appName = useContext(CAppName);
+    let loginStatus = useContext(CLoginStatus);
+
+    return (
+        <>
+            {
+                (loginStatus.isLoggedIn && !loginStatus.needCompleteRegistration) || loginStatus.method === "guest" ?
+                (
+                    props.children
+                ) : (
+                    <Navigate to={`/react/${appName}/login`}/> // TODO @garsuga: abstract
+                )
+            }
+        </>
+    )
+}
+
+type LoginConfirmParams = {
+    token: string,
+    method: AuthMethod,
+    needCompleteRegistration: "0" | "1"
+}
+
+export const LoginConfirm = (props) => {
+    let { token, method, needCompleteRegistration } = useParams<LoginConfirmParams>();
+
+    switch(method) {
+        case 'email':
+            return <LoginEmailConfirm needCompleteRegistration={needCompleteRegistration} token={token}/>
+        default:
+            throw new Error(`could not handle login method=${method}`)
+    }
+}
+
+
+
+export const LoginNeedCompleteRegistration = (props) => {
+    let appName = useContext(CAppName);
+    let navigate = useNavigate();
+    let onCompleteForm = (data: {[key: string]: any}) => {
+        fetch(`/auth-complete-registration`, {
+            method: "POST",
+            body: JSON.stringify({ ...data, simulationType: appName }),
+            headers: {
+                "Content-Type": "application/json"
+            }
+        }).then(() => navigate(`/react/${appName}`))
+    }
+    return (
+        <LoginExtraInfoForm onComplete={(data) => onCompleteForm(data)}/>
+    )
+}
+
+export const LoginExtraInfoForm = (props: { onComplete: ({displayName}) => void }) => {
+    let { onComplete } = props;
+    let [displayName, updateDisplayName] = useState<string>("");
+    return (
+        <>
+            <p>Please enter your full name to complete your Sirepo registration.</p>
+            <Form.Group as={Row} className="mb-3" controlId="formPlaintextEmail">
```

Comment:
I'm concerned about the proliferation of these constants which are implicitly coupled. mb-3 is implicit and so is formPlaintextEmail. These sh